aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,63 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  /* Init to column major format. `sol` and `lu` need to be contiguous
+   * since we pass sol.data() and lu.data() to Lapack */
+  bool tc = sol.dim() == 2 && isTransposeContiguous(sol);
+  if (tc) {
+    // if transpose of output tensor is contiguous, use it
+    sol.t_();
+  }
+
+  sol.resize_({by, bx});
+  if (self.data_ptr() == sol.data_ptr()) {
```

Comment:
If I remove this optimization and always do `sol.copy_(self.view({bx, by}).t().clone()` it will work. This code handles three cases:
1) When `sol` and `self` are the same Tensor (I'll revert to the previous address check)
    a) `sol` is transpose contiguous, so we transpose it (need to transpose input B anyway for Lapack)
    b) `sol`'s transpose is not contiguous, so it should be contiguous, but we need to copy the transpose of the input (`self`) into this contiguous tensor before passing it to Lapack. Since they share underlying storage, `self.view(...).t()` is cloned before copying to sol.
2) When `sol` and `self` are different tensors: we can do the copy to `sol` from `self` without the extra `.clone()`

Reply:
Thanks. I think there might be one case that is not covered. See my other comment.